src/compiler/scala/tools/nsc/typechecker/Namers.scala

```diff
@@ -710,6 +710,9 @@ trait Namers extends MethodSynthesis {
       if (isScala) {
         if (nme.isSetterName(tree.name)) ValOrVarWithSetterSuffixError(tree)
         if (tree.mods.isPrivateLocal && tree.mods.isCaseAccessor) PrivateThisCaseClassParameterError(tree)
+        if (tree.mods.isImplicit && !owner.isTerm && settings.warnImplicitNoAnnotation && tree.tpt.isEmpty) {
+          reporter.warning(tree.pos, s"implicit value ${tree.name} has no type annotation")
```

Comment:
Should be `.name.decoded` to avoid displaying encoded names like '$colon$colon`.

Reply:
Not that further motivation is needed, but I expected it to warn about lacking an implicit not found annotation.